lib/ansible/modules/cloud/amazon/ec2_lc.py

```diff
@@ -250,6 +263,8 @@ def create_launch_config(connection, module):
             if bdm.ebs is not None:
                 result['block_device_mappings'][-1]['ebs'] = dict(snapshot_id=bdm.ebs.snapshot_id, volume_size=bdm.ebs.volume_size)
 
+    if launch_configs[0].user_data != "":
+        result['user_data'] = "hidden" # Otherwise, we dump binary to the user's terminal
```

Comment:
I'd be happy with that method (only show if it's provided inline, not if it's in a file). Pretty simple, and if users have it in a file they can always open it/unzip it/whatever.